<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UI::FilenameTemplateWidget</class>
 <widget class="QWidget" name="FilenameTemplate">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>371</width>
    <height>183</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Setup filename template</string>
  </property>
  <layout class="QVBoxLayout" name="windowLayout">
   <property name="margin">
    <number>2</number>
   </property>
   <item>
    <widget class="QGroupBox" name="dirSelectionGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Select directory</string>
     </property>
     <layout class="QHBoxLayout" name="directoryLayout">
      <property name="margin">
       <number>2</number>
      </property>
      <item>
       <widget class="QComboBox" name="DirectoryName">
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="maxCount">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="browseDirButton">
        <property name="text">
         <string notr="true">...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="nameSelectionGroup">
     <property name="title">
      <string>Specify filename template</string>
     </property>
     <layout class="QVBoxLayout" name="nameSelectionLayout">
      <property name="margin">
       <number>2</number>
      </property>
      <item>
       <widget class="QComboBox" name="FileTemplate">
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="maxCount">
         <number>10</number>
        </property>
        <item>
         <property name="text">
          <string notr="true">[Fullpath].[Type]</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">[Filename]_[Subpath].[Type]</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">[CRC].[Type]</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="hintsLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="hintFullPath">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means full source data path including any nested container subpath normalized to filename.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Example:&lt;/b&gt;
&lt;pre&gt;/path/to/file.ext?nested.bin/+256/subfile.B&lt;/pre&gt; will be translated to
&lt;pre&gt;_path_to_file.ext_nested.bin_+256_subfile.B&lt;/pre&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Fullpath]&quot;&gt;[Fullpath]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="hintPath">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means source file path normalized to filename.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Example:&lt;/b&gt;&lt;/p&gt;
&lt;pre&gt;/path/to/file.ext?nested.bin/+256/subfile.B&lt;/pre&gt; will be translated to
&lt;pre&gt;_path_to_file.ext&lt;/pre&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Path]&quot;&gt;[Path]&lt;/a&gt;&lt;/p&gt;
&lt;/body&gt;
&lt;/html&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="hintFilename">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means source data filename.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Example:&lt;/b&gt;
&lt;pre&gt;/path/to/file.ext?nested.bin/+256/subfile.B&lt;/pre&gt; will be translated to
&lt;pre&gt;file.ext&lt;/pre&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Filename]&quot;&gt;[Filename]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="hintExtension">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means source data filename's extension.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Example:&lt;/b&gt;
&lt;pre&gt;/path/to/file.ext?nested.bin/+256/subfile.B&lt;/pre&gt; will be translated to
&lt;pre&gt;ext&lt;/pre&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Extension]&quot;&gt;[Extension]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="hintSubpath">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means subdata identifier in source file normalized to filename format.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Example:&lt;/b&gt;
&lt;pre&gt;/path/to/file.ext?nested.bin/+256/subfile.B&lt;/pre&gt; will be translated to
&lt;pre&gt;nested.bin_+256_subfile.B&lt;/pre&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Subpath]&quot;&gt;[Subpath]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="hintContainer">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means subdata containers chain identifiers normalized to filename format.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Example:&lt;/b&gt;
&lt;pre&gt;/path/to/file.ext?nested.bin/+256/subfile.B&lt;/pre&gt; can be translated to
&lt;pre&gt;ZIP_RAW_TRD&lt;/pre&gt; in case if file.ext is ZIP file and TRD file is located at 256-byte offset of nested.bin file
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Container]&quot;&gt;[Container]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="hintType">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means chiptune type. Usually 2-3 uppercased letters. Can be used as an extension.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Type]&quot;&gt;[Type]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="hintSize">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means chiptune size in bytes.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Size]&quot;&gt;[Size]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="hintCRC">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means CRC32 checksum of chiptune in decimal form.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[CRC]&quot;&gt;[CRC]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QLabel" name="hintFixedCRC">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means CRC32 checksum if meaningful chiptune parts in decimal form.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[FixedCRC]&quot;&gt;[FixedCRC]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="hintTitle">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means chiptune title and can be empty.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Title]&quot;&gt;[Title]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="hintAuthor">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means chiptune's author and can be empty.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Author]&quot;&gt;[Author]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLabel" name="hintComment">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means chiptune comment and can be empty.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Comment]&quot;&gt;[Comment]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QLabel" name="hintProgram">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means program familiy chiptune was created in or compatible with.&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Program]&quot;&gt;[Program]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QLabel" name="hintVersion">
          <property name="toolTip">
           <string>
&lt;p&gt;This field means program version chiptune was created in. Can be not specified. Format is 10*MAJOR+MINOR&lt;/p&gt;
</string>
          </property>
          <property name="text">
           <string notr="true">&lt;a href=&quot;[Version]&quot;&gt;[Version]&lt;/a&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="parametersGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Specify parameters</string>
     </property>
     <layout class="QHBoxLayout" name="parametersLayout">
      <property name="margin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLabel" name="overwriteExistingLabel">
        <property name="text">
         <string>Existing files</string>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="overwriteExisting">
        <item>
         <property name="text">
          <string>skip</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>overwrite</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>generate new name</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="createDirectories">
        <property name="text">
         <string>Create path</string>
        </property>
        <property name="toolTip">
          <string>Force to create all intermediate directories of target files</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>browseDirButton</sender>
   <signal>clicked()</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnBrowseDirectory()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintFullPath</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintPath</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintFilename</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintExtension</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintSubpath</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintContainer</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintType</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintSize</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintCRC</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintFixedCRC</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintTitle</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintAuthor</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintComment</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintProgram</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintVersion</sender>
   <signal>linkActivated(QString)</signal>
   <receiver>FilenameTemplate</receiver>
   <slot>OnClickHint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnBrowseDirectory()</slot>
  <slot>OnClickHint(QString)</slot>
 </slots>
</ui>
